src/neuroglancer/datasource/graphene/frontend.ts

```diff
@@ -874,10 +873,30 @@ class GrapheneGraphSource extends SegmentationGraphSource {
     return this.graphServer.getRoot(segment);
   }
 
-  tab(layer: SegmentationUserLayer) {
-    return new GrapheneTab(layer);
+  tabContents(layer: SegmentationUserLayer) {
+    return new DependentViewWidget(
+                                layer.displayState.segmentationGroupState.value.graph,
+                                (graph, parent, context) => {
+                                  if (graph === undefined) return;
```

Comment:
I think SegmentationGraphSourceTab should be responsible for only including the tab contents if these conditions hold, so that individual graph sources don't need these checks.  Then you wouldn't need a second layer of dependent view widget.

Reply:
The primary reason I used it was that it was also supplying the context, should I just add context as a parameter? Also perhaps element/tab to solve the css class issue I mentioned above?

```tabContents(layer: SegmentationUserLayer, context: DependentViewContext, tab: SegmentationGraphSourceTab) {```

Reply:
this is what I was thinking https://github.com/google/neuroglancer/pull/395/commits/3cfdc6dda9c9cdfcada5156952ec8245266081da